### PR TITLE
Add single aggregate

### PR DESCRIPTION
**Pull request by kamikaz1k**
Also:
- added a chart "manager" that only adds chart
- stores/retrieves chart information from `localStorage`
- has failure recovery (via state erasure) if `localStorage` is malformed
- removal of charts will be tricky cuz of the way I implemented it...but it works _now_

xharts-single-metric-charts

* **Edit: Now with close button**

xhart-close-chart

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,34 +1,114 @@
...
+  parseQueryParams(paramString) {
+    return paramString.split("&").filter(v => !!v).reduce((params, keyVal) => {
+      const [key, val] = keyVal.split("=");
+      params[key] = val;
+      return params;
+    }, {});
+  }
+
+  getQueryParams() {
+    console.log("query params", window.location.search);
```
Comment (woodgern) [inline src/App.js:54]: Does react really not have something built in that wraps query param functionality?

Comment (kamikaz1k) [reply src/App.js:54]: React isn't a _framework_ it's a **view** library 😆 

I can get ReactRouter but I didn't wanna do that yet....

**Comment by kamikaz1k** [comment]
Race condition because if there's a network request inflight, and then you remove the chart, if the promise isn't cancelled, it'll overwrite data intended for something else.

**Comment by woodgern** [comment]
Guess you could have the delete block on any request with a lock or something, but the reasonable solution seems to just move it all to the backend

**Comment by kamikaz1k** [comment]
I can do checking, but in practice it's fast enough. Also, page refresh fixes it ;)

**Pull request merged**